lib/bolt/transport/api.rb

```diff
@@ -7,21 +7,27 @@
 module Bolt
   module Transport
     class Api < Orch
-      def initialize(*args)
-        super
+      attr_reader :plan_job
+
+      def initialize(plan_job, *args)
+        @plan_job = plan_job
+        @connection = nil
+        super(*args)
       end
 
       def get_connection(conn_opts)
-        key = Bolt::Transport::Api::Connection.get_key(conn_opts)
-        unless (conn = @connections[key])
-          @connections[key] = Bolt::Transport::Api::Connection.new(conn_opts, logger)
-          conn = @connections[key]
-        end
-        conn
+        @connection ||= Bolt::Transport::Api::Connection.new(conn_opts, logger, @plan_job)
       end
 
       def batches(targets)
-        targets.group_by { |target| Bolt::Transport::Api::Connection.get_key(target.options) }.values
+        targets
+      end
+
+      def finish_plan(result)
+        # CODEREVIEW should we check that result is a Bolt::PlanResult?
+        @connection.finish_plan(result)
+      rescue StandardError => e
+        @logger.debug("Failed to finish plan on #{@connection.key}: #{e.message}")
```

Comment:
This seems like a serious error in the case of a service since its means data is probably lost.

Reply:
Logging it at error with the entire `result` seems better.

Reply:
the error message? I think the error message should be updated to include both the error and the entire json format plan result.